go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -0,0 +1,120 @@
+package clustermanager
+
+import (
+	"sync"
+	"time"
+
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// ovnkubeClusterManagerLeaderMetrics object is used for the cluster manager
+// leader election metrics
+type ovnkubeClusterManagerLeaderMetrics struct{}
+
+// On will be called by leader election (LE) when cluster manager becomes a leader
+func (ovnkubeClusterManagerLeaderMetrics) On(string) {
+	metrics.MetricClusterManagerLeader.Set(1)
+}
+
+// On will be called by LE when cluster manager becomes a follower
+func (ovnkubeClusterManagerLeaderMetrics) Off(string) {
+	metrics.MetricClusterManagerLeader.Set(0)
+}
+
+// ovnkubeClusterManagerLeaderMetricsProvider is used by LE
+type ovnkubeClusterManagerLeaderMetricsProvider struct{}
+
+// ovnkubeClusterManagerLeaderMetricsProvider is called by LE to create an instance of
+// ovnkubeClusterManagerLeaderMetrics
+func (_ ovnkubeClusterManagerLeaderMetricsProvider) NewLeaderMetric() leaderelection.SwitchMetric {
+	return ovnkubeClusterManagerLeaderMetrics{}
+}
+
+// ClusterManager structure is the object which manages the cluster nodes.
+// It creates a default network controller for the default network and a
+// secondary network cluster controller manager to manage the multi networks.
+type ClusterManager struct {
+	client                               clientset.Interface
+	defaultNetClusterController          *defaultNetworkClusterController
+	wf                                   *factory.WatchFactory
+	wg                                   *sync.WaitGroup
+	stopChan                             chan struct{}
```

Comment:
We don't have a need for this stopChan then (not used for anything)?

Reply:
Yes.  useless stopChan.  I'll delete it.